File: app-mvvm/src/main/java/com/xuie/imaginaryandroid/gui/nets/video/VideosContract.java
package com.xuie.imaginaryandroid.gui.nets.video;

import com.xuie.imaginaryandroid.BasePresenter;
import com.xuie.imaginaryandroid.BaseView;
import com.xuie.imaginaryandroid.data.VideoBean;

import java.util.List;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface VideosContract {

    interface View extends BaseView {

        void addList(boolean isRefresh, List<VideoBean> videoBeen);
    }

    interface Presenter extends BasePresenter {
        void getList(String type, boolean isRefresh);
    }
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/gui/nets/video/VideosContract.kt
package com.xuie.imaginaryandroid.gui.nets.video

import com.xuie.imaginaryandroid.BasePresenter
import com.xuie.imaginaryandroid.BaseView
import com.xuie.imaginaryandroid.data.VideoBean

/**
 * This specifies the contract between the view and the presenter.
 */
interface VideosContract {

    interface View : BaseView<Presenter> {

        fun addList(isRefresh: Boolean, videoBeen: List<VideoBean>)
    }

    interface Presenter : BasePresenter {
        fun getList(type: String, isRefresh: Boolean)
    }
}
